code/game/machinery/doors/airlock.dm

```diff
@@ -570,10 +570,77 @@ There are 9 wires.
 	else
 		return 0
 
+/obj/machinery/door/airlock/proc/force_wedge_item(obj/item/weapon/tool/T)
+	usr.drop_item()
+	T.forceMove(src)
+	wedged_item = T
+	update_icon()
+	verbs -= /obj/machinery/door/airlock/proc/try_wedge_item
+	verbs += /obj/machinery/door/airlock/proc/take_out_wedged_item
+
+/obj/machinery/door/airlock/proc/try_wedge_item()
+	set name = "Wedge item"
+	set category = "Object"
+	set src in view(1)
+
+	var/obj/item/weapon/tool/T = usr.get_active_hand()
+	if(istype(T) && T.w_class >= ITEM_SIZE_NORMAL) // We do the checks before proc call, because see "proc overhead".
+		if(!density)
+			force_wedge_item(T)
+			to_chat(usr, SPAN_NOTICE("You wedge [T] into [src]."))
+		else
+			to_chat(usr, SPAN_NOTICE("[T] can't be wedged into [src], while [src] is open."))
+
+/obj/machinery/door/airlock/proc/take_out_wedged_item(mob/living/user)
+	set name = "Take wedged item out"
```

Comment:
"Take wedged item out" is a somewhat overlong and clumsy name for a verb, please revise

Possible alternative, Remove Blockage

Reply:
"Unjam Airlock" "Unwedge Airlock"